Common/Data/Custom/PsychSignal/PsychSignalSentimentData.cs

```diff
@@ -0,0 +1,137 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Globalization;
+using System.IO;
+
+namespace QuantConnect.Data.Custom.PsychSignal
+{
+    public class PsychSignalSentimentData : BaseData
+    {
+        /// <summary>
+        /// Bullish intensity as reported by psychsignal
+        /// </summary>
+        public decimal BullIntensity;
+
+        /// <summary>
+        /// Bearish intensity as reported by psychsignal.
+        /// </summary>
+        public decimal BearIntensity;
+
+        /// <summary>
+        /// Bullish intensity minus bearish intensity
+        /// </summary>
+        public decimal BullMinusBear;
+
+        /// <summary>
+        /// Total bullish scored messages. 
+        /// This is the total nubmer of messages classified as bullish in a minute
+        /// </summary>
+        public int BullScoredMessages;
+
+        /// <summary>
+        /// Total bearish scored messages. 
+        /// This is the total number of messages classified as bearish in a minute
+        /// </summary>
+        public int BearScoredMessages;
+
+        /// <summary>
+        /// Bull/Bear message ratio.
+        /// Calculated by dividing BullScoredMessages by BearScoredMessages
+        /// </summary>
+        /// <remarks>If bearish messages equals zero, then the resulting value equals zero</remarks>
+        public decimal BullBearMessageRatio;
+
+        /// <summary>
+        /// Total messages scanned.
+        /// This includes bull/bear messages and messages that couldn't be classified
+        /// </summary>
+        /// <remarks>
+        /// Sometimes, there will be no bull/bear rated messages, but nonetheless had messages scanned.
+        /// This field describes the total fields that were scanned in a minute
+        /// </remarks>
+        public int TotalScoredMessages;
+
+        /// <summary>
+        /// Retrieve Psychsignal data from disk and return it to user's custom data subscription
+        /// </summary>
+        /// <param name="config">Configuration</param>
+        /// <param name="date">Date of this source file</param>
+        /// <param name="isLiveMode">true if we're in livemode, false for backtesting mode</param>
+        /// <returns></returns>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            return new SubscriptionDataSource(
+                Path.Combine(
+                    Globals.DataFolder,
+                    "alternative",
+                    "psychsignal",
+                    $"{config.Symbol.Value.ToLower()}",
+                    $"{date:yyyyMMdd}.zip#{date:yyyyMMdd}.csv"
```

Comment:
I _think_ that if the zip just has one entry specifying it is not required https://github.com/QuantConnect/Lean/blob/master/Engine/DataFeeds/ZipDataCacheProvider.cs#L205